graphql-native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ResponseGenerator.java

```diff
@@ -54,32 +59,42 @@
     private ResponseGenerator() {
     }
 
-    static void getDataFromResult(Environment environment, BObject visitor, BObject node, Object result,
-                                  BMap<BString, Object> data, CallbackHandler callbackHandler) {
-        if (result instanceof BValue) {
-            int tag = ((BValue) result).getType().getTag();
-            if (tag < TypeTags.JSON_TAG) {
-                data.put(node.getStringValue(NAME_FIELD), result);
-            } else if (tag == TypeTags.RECORD_TYPE_TAG) {
-                getDataFromRecord(environment, visitor, node, (BMap<BString, Object>) result, data, callbackHandler);
-            } else if (tag == TypeTags.MAP_TAG) {
-                getDataFromMap(environment, visitor, node, (BMap<BString, Object>) result, data, callbackHandler);
-            } else if (tag == TypeTags.ARRAY_TAG) {
-                getDataFromArray(environment, visitor, node, (BArray) result, data, callbackHandler);
-            } else if (tag == TypeTags.TABLE_TAG) {
-                getDataFromTable(environment, visitor, node, (BTable) result, data, callbackHandler);
-            } else if (tag == TypeTags.SERVICE_TAG) {
-                getDataFromService(environment, (BObject) result, visitor, node, data, new ArrayList<>(),
-                                   callbackHandler);
-            } // Here, `else` should not be reached.
+    static void populateDocument(Environment environment, BObject visitor, BObject node, Object result,
```

Comment:
I think `populateResponse` makes more sense here. `populateDocument` implies that this is populating the document, which we should not be doing.